### PR TITLE
NoJira: Multiple set statements do not work in mongo, so rework the sift update

### PR DESCRIPTION
**Pull request by benjstephenson**
BSON to ensure all sub-docs are correctly updated.

### CODE REVIEW

File: app/services/sift/ApplicationSiftService.scala [hunk 1]
```diff
@@ -76,28 +76,27 @@ trait ApplicationSiftService extends CurrentSchemeStatusHelper with CommonBSONDo
...
       val siftedSchemes = (currentSiftEvaluation.map(_.schemeId) :+ result.schemeId).distinct
 
-      val mergedUpdate = Seq(
-        BSONDocument("$set" -> currentSchemeStatusBSON(newSchemeStatus)),
+      val settableFields = Seq(currentSchemeStatusBSON(newSchemeStatus),
         maybeSetProgressStatus(siftedSchemes.toSet, candidatesSiftableSchemes.toSet)
-      ).foldLeft(siftBson) { (acc, doc) => acc ++ doc }
+      )
+
+      play.api.Logger.error(s"\n\nBLAH : ${settableFields.map(BSONDocument.pretty)}")
```
Comment (vglushak) [inline app/services/sift/ApplicationSiftService.scala]: BLAH? is that 'Bringing Life And Hope' or 'Boring Links and Homepages'? :)

Comment (benjstephenson) [inline app/services/sift/ApplicationSiftService.scala]: Oops, I'll remove that!

**Comment by vglushak** [comment]
I'll merge as soon as checks passed.

File: test/services/sift/ApplicationSiftServiceSpec.scala [hunk 2]
```diff
@@ -65,11 +65,8 @@ class ApplicationSiftServiceSpec extends ScalaMockUnitSpec {
 
   trait SiftUpdateTest extends TestFixture {
     val progressStatusUpdateBson = BSONDocument(
-      "$set" -> BSONDocument(
-        s"progress-status.${ProgressStatuses.SIFT_COMPLETED}" -> true,
-        s"progress-status-timestamp.${ProgressStatuses.SIFT_COMPLETED}" ->
-          BSONDateTimeHandler.write(DateTimeFactoryMock.nowLocalTimeZone)
-      )
+      s"progress-status.${ProgressStatuses.SIFT_COMPLETED}" -> true,
```
Comment (henricook) [inline test/services/sift/ApplicationSiftServiceSpec.scala]: Is this correct? It looks like it's no longer doing a set and is doing an overwrite

Comment (henricook) [inline test/services/sift/ApplicationSiftServiceSpec.scala]: Technically i think we're meant to be doing  { progress-status: { ProgressStatuses.SIFT_COMPLETED: true } } or the keys set all wonky, but i think we're doing that wrong *everywhere*

Comment (benjstephenson) [inline test/services/sift/ApplicationSiftServiceSpec.scala]: Yup, the repo method that this service calls will wrap all the fields that are to be set in a single $set operator

**Pull request merged**